test/test_autograd.py

```diff
@@ -2843,6 +2843,14 @@ def test_pow_scalar_base(self):
         a = torch.arange(1, 13, dtype=torch.double).view(3, 4).requires_grad_()
         gradcheck(lambda a: torch.pow(2, a), (a,))
 
+    def test_igamma(self):
+        # 1e-3 offset to avoid zeros
+        # NOTE: derivative for s is not implemented
+        s = (torch.rand(3, dtype=torch.double) + 1e-3)
+        x = (torch.rand(3, dtype=torch.double) + 1e-3).requires_grad_()
```

Comment:
Consider testing on more points, say 100 points instead of 3. As I understand this is basically free due to tensor overhead.

Reply:
Done. I've added the tests for 100 points and it failed, so I just changed it to follow scipy's implementation (which makes the code substantially larger)